packages/scheduler/src/Scheduler.js

```diff
@@ -152,6 +190,8 @@ function flushWork(hasTimeRemaining, initialTime) {
     }
     // Return whether there's additional work
     if (currentTask !== null) {
+      markSchedulerSuspended(currentTime);
+      isHostCallbackScheduled = true;
```

Comment:
I've found this factoring very counterintuitive.

It's like `Scheduler` knows too much about the host config. It was surprising to see `isHostCallbackScheduled = true` here without any requests, and only later realize it "knows" host callback will be scheduled because it returned `true`, and it trusts host config to respect that.

There's similar implicitness in the error code path.

Reply:
The error code path seems inconsistent with what this code seems to imply.

The host config _does_ schedule a callback in the error case (if this throws).
But the `catch` below doesn't have the corresponding lines:

```js
            markSchedulerSuspended(currentTime);
            isHostCallbackScheduled = true;
```

Should it?